### PR TITLE
ISPN-7153 Add documentation regarding memory based eviction

### PR DESCRIPTION
**Pull request by wburns**
https://issues.jboss.org/browse/ISPN-7153

### CODE REVIEW

File: documentation/src/main/asciidoc/user_guide/eviction.adoc [hunk 1]
```diff
@@ -1,16 +1,24 @@
...
 
-IMPORTANT: that eviction occurs on a _local_ basis, and is not cluster-wide.  Each node runs an eviction thread to analyse the contents of its in-memory container and decide what to evict. Eviction does not take into account the amount of free memory in the JVM as threshold to  starts evicting entries. You have to set maxEntries attribute of the eviction element to be greater than zero in order for eviction to be turned on. If maxEntries is too large you can run out of memory. maxEntries attribute will probably take some tuning in each use case.
+IMPORTANT: that eviction occurs on a _local_ basis, and is not cluster-wide.  Each node runs an eviction thread to analyse the contents of its in-memory container and decide what to evict. Eviction does not take into account the amount of free memory in the JVM as threshold to  starts evicting entries. You have to set size attribute of the eviction element to be greater than zero in order for eviction to be turned on. If size is too large you can run out of memory. size attribute will probably take some tuning in each use case.
```
Comment (danberindei) [inline documentation/src/main/asciidoc/user_guide/eviction.adoc]: Could you split this into multiple lines and use backticks around `size`?

File: documentation/src/main/asciidoc/user_guide/eviction.adoc [hunk 2]
```diff
@@ -87,19 +109,38 @@ Programmatically, the same would be defined using:
...
+               .size(2000).expiration().wakeUpInterval(5000l).lifespan(1000l).maxIdle(500l)
                .build();
 
 ----
 
+==== Memory Based Eviction Configuration
+Memory based eviction may require some additional configuration options if you
+are using your own custom types as Infinispan is normally used.  In this case
+Infinispan cannot estimate the memory usage of your classes and as such you are
+required to use `storeAsBinary` when memory based eviction is used.
```
Comment (danberindei) [inline documentation/src/main/asciidoc/user_guide/eviction.adoc]: * Start sentence on new line.
* Parenthesis around (as Infinispan is normally used)?
* "you are required to use `storeAsBinary" is a bit vague, do we log a warning or throw an exception when an application inserts a custom type in a cache with memory based eviction, or is it just a risk of OOME?
* I'd change the examples to include `storeAsBinary` as well.

File: documentation/src/main/asciidoc/user_guide/eviction.adoc [hunk 3]
```diff
@@ -1,16 +1,24 @@
...
+Eviction is configured by adding the
+link:http://docs.jboss.org/infinispan/{infinispanversion}/configdocs/infinispan-config-{infinispanversion}.html[`<eviction />`]
+element to your `<*-cache />` configuration sections or using
+link:{javadocroot}/org/infinispan/configuration/cache/EvictionConfigurationBuilder.html[EvictionConfigurationBuilder]
+API programmatic approach.
 
 All cache entry are evicted by piggybacking on user threads that are hitting the cache.
-Periodic pruning of expired cache entries from cache is done on a dedicated thread which is turned on by enabling reaper in expiration configuration element/API.
+Periodic pruning of expired cache entries from cache is done on a dedicated thread
+which is turned on by enabling reaper in expiration configuration element/API.
```
Comment (danberindei) [inline documentation/src/main/asciidoc/user_guide/eviction.adoc]: Shouldn't this be moved to the expiration section? Here maybe we should discuss just the interaction between expiration and eviction, e.g. how a non-expired entry could be evicted even though there are expired entries in the cache.

Comment (wburns) [inline documentation/src/main/asciidoc/user_guide/eviction.adoc]: Yeah I wasn't trying to change the overall flow of the document at all.  But I can move these over.

File: documentation/src/main/asciidoc/user_guide/eviction.adoc (hunk 2 above)
Comment (wburns) [inline documentation/src/main/asciidoc/user_guide/eviction.adoc]: 1. Updated all of the paragraphs to have new line
2. Sure
3. It throws an exception that isn't the clearest, but gets the job done :P
4. Added.

**Comment by wburns** [comment]
Updated

**Comment by galderz** [comment]
Integrated

**Pull request closed without merging**